reframe/core/pipeline.py

```diff
@@ -1196,22 +1196,18 @@ def compile(self):
                 else:
                     self.build_system = 'Make'
 
-            self.build_system.srcdir = self.sourcepath
+            if self.sourcepath:
+                self.build_system.srcdir = self.sourcepath
+            else:
+                self.build_system.srcdir = self._stagedir
```

Comment:
As I said revert this and just move the part where we are setting the `build_commands` inside the `osext.change_dir()` a bit later on.